core/src/test/java/org/apache/brooklyn/core/entity/AbstractApplicationLegacyTest.java

```diff
@@ -92,8 +94,11 @@ public void testStartAndStopUnmanagedAppAutomanagesTheAppAndChildren() throws Ex
         assertTrue(Entities.isManaged(app2));
         assertTrue(Entities.isManaged(child));
         assertEquals(child.getCallHistory(), ImmutableList.of("start"));
-        assertEquals(mgmt.getEntityManager().getEntity(app2.getId()), app2);
```

Comment:
Wow, that's a really interesting error (that it used to pass because of `AbstractEntity.equals` matching on `o == selfProxy`, because `this.selfProxy` was null!). Well found and fixed!